src/libs/actions/IOU.js

```diff
@@ -13,7 +16,117 @@ function getPreferredCurrency() {
+            shouldLoadOptionalKeys: true,
+            includePinnedReports: true,
+        }))
+        .then((response) => {
+            if (response.jsonCode !== 200) {
+                throw new Error(response.message);
+            }
+
+            const iouReportData = response.reports[iouReportID];
+            if (!iouReportData) {
+                throw new Error(`No iouReportData found for reportID ${iouReportID}`);
+            }
+
+            Onyx.merge(`${ONYXKEYS.COLLECTION.REPORT_IOUS}${iouReportID}`,
+                getSimplifiedIOUReport(iouReportData));
+            Onyx.merge(ONYXKEYS.IOU, {loading: false, creatingIOUTransaction: false, error: false});
+        })
+        .catch((error) => {
+            Onyx.merge(ONYXKEYS.IOU, {loading: false, creatingIOUTransaction: false, error: true});
+            throw new Error(`[Report] Failed to populate IOU Collection: ${error.message}`);
+        });
+}
+
+/**
+ * Creates IOUSplit Transaction
+ * @param {Object} parameters
+ * @param {Array} parameters.participants
+ * @param {Array} parameters.splits
+ * @param {String} parameters.comment
+ * @param {String} parameters.amount
+ * @param {String} parameters.currency
+ */
+function createIOUSplit({
+    participants,
+    comment,
+    amount,
+    currency,
+    splits,
+}) {
+    let reportIDs = [];
+    Onyx.merge(ONYXKEYS.IOU, {loading: true, creatingIOUTransaction: true, error: false});
+    API.CreateChatReport({
+        emailList: participants.join(','),
+    })
+        .then(data => data.reportID)
+        .then(reportID => API.CreateIOUSplit({
+            splits: JSON.stringify(splits),
+            currency,
+            amount,
+            comment,
+            reportID,
+        }))
+        .then((data) => {
+            reportIDs = data.reportIDList;
+            return reportIDs;
+        })
+        .then(reportIDList => API.Get({
+            returnValueList: 'reportStuff',
+            reportIDList,
+            shouldLoadOptionalKeys: true,
+            includePinnedReports: true,
+        }))
+        .then(({reports}) => _.map(reports, getSimplifiedIOUReport))
+        .then((iouReportObjects) => {
+            const reportIOUData = {};
+            _.each(iouReportObjects, (iouReportObject) => {
+                if (!iouReportObject) {
+                    return;
+                }
+                const iouReportKey = `${ONYXKEYS.COLLECTION.REPORT_IOUS}${iouReportObject.reportID}`;
+                reportIOUData[iouReportKey] = iouReportObject;
+            });
+            return Onyx.mergeCollection(ONYXKEYS.COLLECTION.REPORT_IOUS, {...reportIOUData});
+        })
+        .then(() => {
+            Onyx.merge(ONYXKEYS.IOU, {loading: false, creatingIOUTransaction: false, error: false});
+        })
+        .catch((error) => {
+            console.debug(`Error: ${error.message}`);
+            Onyx.merge(ONYXKEYS.IOU, {loading: false, creatingIOUTransaction: false, error: true});
+        });
```

Comment:
This can be simplified a bit.

`then`: can be removed
`catch`: can replaced with `Onyx.merge(ONYXKEYS.IOU, {error: true});`
then, add a `finally`:   `Onyx.merge(ONYXKEYS.IOU, {loading: false, creatingIOUTransaction: false});`